pkg/kubelet/kubelet_network.go

```diff
@@ -31,7 +31,7 @@ func (kl *Kubelet) providerRequiresNetworkingConfiguration() bool {
 	// is used or whether we are using overlay networking. We should return
 	// true for cloud providers if they implement Routes() interface and
 	// we are not using overlay networking.
-	if kl.cloud == nil || kl.cloud.ProviderName() != "gce" {
+	if kl.cloud == nil {
```

Comment:
I think that old comment // TODO is no longer valid. we are checking routes just right after. Can we remove?

As for the change itself: I don't know why there was a special case for a specific cloud provider. But maybe @bowei et al may know.

Reply:
https://github.com/kubernetes/kubernetes/blob/bad4faf1b9d69d3cb6e8de90dc5eee1d8cf0b6d3/pkg/kubelet/kubelet_node_status.go#L336-L345

This is the only location that uses this method
According to my understanding, this condition is required only for GCE

FYI